enrichments/trace/internal/elastic/span.go

```diff
@@ -189,6 +189,7 @@ func (s *spanEnrichmentContext) Enrich(span ptrace.Span, cfg config.Config) {
 func (s *spanEnrichmentContext) enrich(span ptrace.Span, cfg config.Config) {
 	if s.isTransaction {
 		s.enrichTransaction(span, cfg.Transaction)
+		s.enrichExitSpanTransaction(span, cfg)
```

Comment:
Feel free to ignore, maybe this is personal preference. But the code may be a bit easier to follow when pulling in the if condition to this method, so that it's clear that only transactions that are of type client or producer get enriched.

Reply:
I'm wondering whether we need a dedicated method for this at all.

Couldn't we just have some heuristic method in addition that just identifies whether a span is an exist span, use that in an additional condition here, but then just calling the existing `enrichSpan()` method.

Reply:
I still needed to extend `enrichSpan()` a bit, but now I got rid of `enrichExitSpanTransaction` - so there is only `enrichSpan` and `enrichTransaction` as we had it originally.

The 2 main differences in `enrichExitSpanTransaction` where:
- Not setting `processor.event` to `span`
- Setting `transaction.type`

We handle these now with a condition.